tests/test_utils.py

```diff
@@ -0,0 +1,110 @@
+import uuid
+import time
+import pytest
+
+from multiprocessing import Process
+from redis import Redis
+from rq import Queue, Worker, push_connection, pop_connection
+from rq.command import send_shutdown_command
+from creator.utils import stop_job
+
+
+TASK_DURATION = 2
+
+
+def clean_up(queues):
+    """
+    Delete jobs from all queues, and registries. Shutdown all workers for each
+    queue
+    """
+    for q in queues:
+        for job in q.jobs:
+            job.delete()
+        for worker in Worker.all(queue=q):
+            send_shutdown_command(q.connection, worker.name)
+
+
+@pytest.fixture
+def queues():
+    # Setup redis, queues, and workers
+    conn = Redis()
+    push_connection(conn)
+    queues = [Queue(name=str(uuid.uuid4())) for i in range(2)]
+
+    # Cleanup
+    clean_up(queues)
+
+    yield queues
+
+    # Cleanup
+    clean_up(queues)
+    pop_connection()
+
+
+def task(task_name, duration=TASK_DURATION):
+    """
+    Long running task for queue_name
+    """
+    print(f"Starting long task: {task_name}")
+
+    t_end = time.time() + duration
+    progress = ''
+    while time.time() < t_end:
+        progress += '.'
+        print(progress)
+        time.sleep(2)
+
+    print(f"Finished long task: {task_name}")
+
+    return task
+
+
+def enqueue_all(queues):
+    """
+    Submit a task to all queues
+    """
+    for queue in queues:
+        task_name = queue.name + "-task"
+        job_id = queue.name + "-job"
+        queue.enqueue(task, args=(task_name,), job_id=job_id)
+
+
+def work_and_kill(worker, *args, **kwargs):
+    """
+    Do work in another proces and then kill job
+    """
+    def make_stop_job_func(*args, **kwargs):
+        def _stop_job():
+            """
+            Wait for half task duration, then kill job
+            """
+            time.sleep(TASK_DURATION - (TASK_DURATION)/2)
+            stop_job(*args, **kwargs)
+        return _stop_job
+
+    p = Process(target=make_stop_job_func(*args, **kwargs))
+    p.start()
+    worker.work(burst=True)
+    p.join(1)
+
+
+@pytest.mark.parametrize(
+    'delete',
+    [False, True]
+)
+def test_stop_job(queues, delete):
```

Comment:
This seems like quite a complicated test setup. Think you could do the same with much less work by mocking out the response from `get_status()` and `fetch_job()` and also test more cases more easily and spend less time in the actual test execution.

Reply:
Hmm are those the bottlenecks? I think its my task which sleeps for 2 seconds. So the truth is I didn't want to deal with mocking things :) I just thought it would be easier but also more thorough to do real calls